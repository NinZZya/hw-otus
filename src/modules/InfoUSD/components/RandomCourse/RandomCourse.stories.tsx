import React from "react";
// also exported from "@storybook/react" if you can deal with breaking changes in 6.1
import { Meta } from "@storybook/react/types-6-0";

import { RandomCourse } from "./RandomCourse";

export default {
  title: "Modules/InfoUSD/Components/RandomCourse",
  component: RandomCourse,
} as Meta;

export const WithValue = (): React.ReactNode => <RandomCourse />;
