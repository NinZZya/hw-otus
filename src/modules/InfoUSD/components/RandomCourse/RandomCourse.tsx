import React from "react";
import { Card, Indicator, Label } from "../../../../componets";
import { getRandomColor } from "../../../../utils/utils";

const isUPressed = (evt: KeyboardEvent): boolean =>
  evt.key === "U" || evt.key === "u";

interface RandomCourseProp {
  currency: string;
  onGetCourse: () => Promise<number>;
}

interface RandomCourseState {
  value: number;
  color: string;
}

export class RandomCourse extends React.Component<
  RandomCourseProp,
  RandomCourseState
> {
  constructor(props: RandomCourseProp) {
    super(props);
    this.state = {
      value: 0,
      color: "#ff0000",
    };

    this.handleUPressed = this.handleUPressed.bind(this);
  }

  private handleUPressed(evt: KeyboardEvent): void {
    const { onGetCourse } = this.props;
    if (isUPressed(evt)) {
      onGetCourse().then((value: number) => {
        this.setState({
          value,
          color: getRandomColor(),
        });
      });
    }
  }

  componentDidMount(): void {
    document.addEventListener("keydown", this.handleUPressed);
  }

  componentWillUnmount(): void {
    document.removeEventListener("keydown", this.handleUPressed);
  }

  render(): React.ReactNode {
    const { currency } = this.props;
    const { value, color } = this.state;

    return (
      <Card>
        <Indicator type={"card"} color={color} />
        <span>{`For get Random Course from server press "U":`}</span>
        <Label value={`${currency}${value}`} />
      </Card>
    );
  }
}
