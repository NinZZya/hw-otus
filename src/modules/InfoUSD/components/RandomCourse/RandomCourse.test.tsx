import React from "react";
import renderer from "react-test-renderer";
import { RandomCourse } from "./RandomCourse";

it("RandomCourse render coorectly", (): void => {
  const tree = renderer.create(<RandomCourse />).toJSON();

  expect(tree).toMatchSnapshot();
});
