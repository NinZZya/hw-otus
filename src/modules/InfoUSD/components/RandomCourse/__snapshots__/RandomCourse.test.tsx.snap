// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomCourse render coorectly 1`] = `
<div
  className="css-1y7yulv"
>
  <span
    className="css-21d7o5"
    color="#ff0000"
  >
    â€¢
  </span>
  <span>
    For get Random Course from server press "U":
  </span>
  <span
    className="css-l3kone"
  >
    $0
  </span>
</div>
`;
